chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -320,50 +320,59 @@ trait VecLike[T <: Data] extends collection.IndexedSeq[T] with HasId {
     SeqUtils.oneHotMux(indexWhereHelper(p))
 }
 
-/** Base class for data types defined as a bundle of other data types.
+/** Base class for Aggregates based on pairs of (String, Data)
   *
-  * Usage: extend this class (either as an anonymous or named class) and define
-  * members variables of [[Data]] subtypes to be elements in the Bundle.
+  * Only used to share implementation code, not exposed as part of API
   */
-class Bundle extends Aggregate {
+abstract class Record extends Aggregate {
+
+  def elements: ListMap[String, Data]
```

Comment:
This should be val elements.

Ideally even scala.immutable.ListMap[String, Data] but that might not be feasible atm given other weirdness in current Bundle.